build_tools/benchmarks/export_benchmark_config.py

```diff
@@ -57,11 +82,26 @@ def main(args: argparse.Namespace):
 
   target_device_names = (set(args.target_device_names)
                          if args.target_device_names is not None else None)
+  device_spec_matchers = None
+  if args.benchmark_presets is not None:
+    device_spec_matchers = []
+    for preset in args.benchmark_presets:
+      matcher = BENCHMARK_PRESET_DEVICE_SPEC_MATCHERS.get(preset)
+      if matcher is None:
+        raise ValueError(f"Unrecognized benchmark preset: '{preset}'.")
+      device_spec_matchers.append(matcher)
+
   grouped_run_config_map = collections.defaultdict(list)
   for run_config in all_run_configs:
     device_name = run_config.target_device_spec.device_name
-    if target_device_names is None or device_name in target_device_names:
-      grouped_run_config_map[device_name].append(run_config)
+    if (target_device_names is not None and
+        device_name not in target_device_names):
+      continue
+    if (device_spec_matchers is not None and
+        all(not matcher(run_config.target_device_spec)
+            for matcher in device_spec_matchers)):
```

Comment:
I think not any would be clearer here, personally:

```suggestion
        not any(matcher(run_config.target_device_spec)
            for matcher in device_spec_matchers)):
```